### PR TITLE
Finished authentication, require login and nickname to comment

### PR DESCRIPTION
**Pull request by joeychin01**
Added nickname, require login and nickname to comment

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -94,4 +94,18 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.setCharacterEncoding("UTF-8");
     response.getWriter().println(gson.toJson(comments));
   }
+
+  /** Function returns nickname from input user id -- maybe put this as a function in comment servlet? */
+  public String getNickname(String id) {
```
Comment (linxiz) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:99]: does this method need to be public?

File: portfolio/src/main/java/com/google/sps/servlets/setNicknameServlet.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import com.google.gson.Gson;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.HashMap;
+
+/** Servlet to set nickname */
+@WebServlet("/set-nick")
+public class setNicknameServlet extends HttpServlet {
```
Comment (linxiz) [inline portfolio/src/main/java/com/google/sps/servlets/setNicknameServlet.java:34]: class name should start with upper case

File: portfolio/src/main/webapp/index.html [hunk 3]
```diff
@@ -32,7 +30,7 @@ <h2>About me</h2>
           My name is Joseph Chinoski, and I am a rising junior at The University of Michigan (Go Blue!).
           At UMich, I am a Computer Science Major in the College of Engineering.
           Besides coding, I play tennis, read books (especially Stephen King), and play the violin!
-            
+
```
Comment (linxiz) [inline portfolio/src/main/webapp/index.html:33]: nit: is this change intended?

File: portfolio/src/main/webapp/index.html [hunk 4]
```diff
@@ -56,23 +54,21 @@ <h2>Experiences</h2>
...
       </div>
 
-
       <div class = "content">
         <p>
           At CSSI I worked in a group of three in order to build a website to help in daily life.
           Through that project, my group built the website lyric-giraffe.appspot.com, a website that 
           displays lyrics to songs, either through directly searching, or by connecting to a user's 
-         Spotify account.
+        Spotify account.
```
Comment (linxiz) [inline portfolio/src/main/webapp/index.html:62]: indentation?

File: portfolio/src/main/webapp/nickname.html [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    
+    <div id="content">
+      <form id="set-nickname" action="/set-nick" method="POST">
+        <p>Please enter a nickname for commenting!</p>
+        <p>This nickname will be associated with your email</p>
+        <p>WARNING: YOUR CHOICE CANNOT BE CHANGED LATER</p>
+        <input type="text" id="nick-choice" name="nick-choice" placeholder="Choose a nickname">
+        <input type="submit">
+      </form>
+    </div>
```
Comment (linxiz) [inline portfolio/src/main/webapp/nickname.html:25]: what is the expected behavior if user has not logged in?

File: portfolio/src/main/webapp/index.html (hunk 3 above)
Comment (joeychin01) [reply portfolio/src/main/webapp/index.html:33]: No it is not

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (joeychin01) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:99]: It does not

File: portfolio/src/main/webapp/nickname.html (hunk 5 above)
Comment (joeychin01) [reply portfolio/src/main/webapp/nickname.html:25]: They cannot set a nickname without logging in, and if they are not logged in they are sent back to the main page

File: portfolio/src/main/webapp/index.html [hunk 6]
```diff
@@ -8,18 +8,16 @@
...
 
   <body onload="getMessages()"> 
-    <button id = "delete-button">DELETE ALL COMMENTS</button>
     <div id="webheader">
       <a href="/"><img src="/images/my_face.jpeg"></a>
       <h1>Joseph Chinoski's Portfolio</h1>
       <a href="https://www.linkedin.com/in/joseph-chinoski/"><img src="/images/LinkedIn-Logo.png"></a>
     </div>
-
+
```
Comment (linxiz) [inline portfolio/src/main/webapp/index.html:16]: remove the spaces if not intended

File: portfolio/src/main/webapp/script.js [hunk 7]
```diff
@@ -51,20 +51,56 @@ function createCommentElement(comment) {
...
 
+/** Tells the server to delete the comment. */
+function deleteComment(comment) {
+  const params = new URLSearchParams();
+  params.append('id', comment.id);
+  fetch('/delete-comment', {method: 'POST', body: params});
+}
+
+
+
```
Comment (linxiz) [inline portfolio/src/main/webapp/script.js:103]: nit: remove extra blank lines

File: portfolio/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+import javax.servlet.http.HttpServletResponse;
+
+/** Deletes a single comment from datastore */
+@WebServlet("/delete-comment")
+public class DeleteCommentServlet extends HttpServlet {
+
+  /** Deletes a single comment from datastore */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```
[the remark this answers is not in the document]
Comment (joeychin01) [reply portfolio/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java:34]: Good idea!
